lib/active_record/open_tracing/processor.rb

```diff
@@ -106,36 +108,51 @@ def sanitize_sql(sql)
         sanitizer ? sanitizer.sanitize(sql) : sql
       end
 
-      def peer_address_tag(config)
-        @peer_address_tag ||= [
+      def peer_address_tag(config, db_instance)
+        [
           "#{config.fetch(:adapter)}://",
           config[:username],
           config[:host] && "@#{config[:host]}",
           "/#{db_instance}"
         ].join
       end
 
-      def db_instance
-        @db_instance ||= connection_config.fetch(:database)
-      end
-
       # If a connection is passed in, pull the db config options hash from it.  This connection
       # is likely to be attached to the query/query client.  Fall back to the application level
       # config.  This change will show writer vs reader/replica dbs in spans when it is determined
       # at the query level.
       def connection_config(connection = nil)
-        unless defined? @connection_config
-          if connection.raw_connection && connection.raw_connection.respond_to?(:query_options)
-            # you have a mysql client
-            @connection_config = connection.raw_connection.query_options
-          else
-            # Rails 6.2 will deprecate ActiveRecord::Base.connection_config
-            @connection_config = ActiveRecord::Base.try(:connection_db_config)&.configuration_hash ||
-                                  ActiveRecord::Base.connection_config
-          end
+        if connection && connection.raw_connection && connection.raw_connection.respond_to?(:query_options)
+          # you have a mysql client
+          connection.raw_connection.query_options
+        else
+          # Rails 6.2 will deprecate ActiveRecord::Base.connection_config
+          ActiveRecord::Base.try(:connection_db_config)&.configuration_hash ||
+                                ActiveRecord::Base.connection_config
         end
+      end
 
-        @connection_config
+      DB_NAME_TO_ROLE = {
+        "primary" => "writing",
```

Comment:
Do you think it would be better to keep this as primary/replica because a lot of apps do a lot of reading from the primary so it might just be a little weird to see the role as `writing` for `select` statements

Reply:
We don't have much control over that actually.

> By default Rails expects the database roles to be writing and reading for the primary and replica respectively.

https://guides.rubyonrails.org/active_record_multiple_databases.html